waku/v2/node/wakunode2.go

```diff
@@ -370,9 +373,28 @@ func (w *WakuNode) Broadcaster() v2.Broadcaster {
 	return w.bcaster
 }
 
-func (w *WakuNode) mountRelay(opts ...pubsub.Option) error {
+func (w *WakuNode) Publish(msg *pb.WakuMessage) error {
+	hash, _ := msg.Hash()
+
+	err := try.Do(func(attempt int) (bool, error) {
+		var err error
+		if w.lightPush.IsStarted() || !w.relay.EnoughPeersToPublish() {
```

Comment:
This logic seems wrong to me. As we have no guarantee relay has been started.
wdyt?